Modules/_testinternalcapi.c

```diff
@@ -102,6 +102,43 @@ test_popcount(PyObject *self, PyObject *Py_UNUSED(args))
 }
 
 
+static int
+check_bit_length(unsigned long x, int expected)
+{
+    int len = _Py_bit_length(x);
+    if (len != expected) {
+        PyErr_Format(PyExc_AssertionError,
+                     "_Py_bit_length(%lu) returns %i, expected %i",
+                     x, len, expected);
+        return -1;
+    }
+    return 0;
+}
+
+
+static PyObject*
+test_bit_length(PyObject *self, PyObject *Py_UNUSED(args))
+{
+#define CHECK(X, RESULT) \
+    do { \
+        if (check_bit_length(X, RESULT) < 0) { \
+            return NULL; \
+        } \
+    } while (0)
+
+    CHECK(0, 0);
+    CHECK(1, 1);
+    CHECK(0x08080808, 28);
+    CHECK(0x10101010, 29);
+    CHECK(0x10204080, 29);
+    CHECK(0xDEADCAFE, 32);
```

Comment:
I suggest other tests:

```suggestion
    CHECK(0x1000, 12);
    CHECK(0x1234, 12);
    CHECK(0x54321, 19);    
    CHECK(0x7FFFFFFF, 31);    
```